xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/v2/CoinmateStreamingAdapter.java

```diff
@@ -71,12 +71,27 @@ public static OpenOrders adaptWebsocketOpenOrders(
                   .timestamp(
                       Date.from(Instant.ofEpochMilli(coinmateWebsocketOpenOrder.getTimestamp())))
                   .limitPrice(BigDecimal.valueOf(coinmateWebsocketOpenOrder.getPrice()))
+                      .orderStatus(statusFromString(coinmateWebsocketOpenOrder.getOrderChangePushEvent()))
                   .build());
         });
     return new OpenOrders(openOrders);
   }
 
-  public static Trade adaptTrade(CoinmateWebSocketTrade webSocketTrade, CurrencyPair currencyPair) {
+    private static Order.OrderStatus statusFromString(String orderChangePushEvent) {
```

Comment:
can you add a null check here and if so return unknown status

Reply:
Yes I can rename, but what do you mean by null check? There's fallback return in the method at the end, that returns UNKNOWN for everything else including null, doesn't it? Or what null check?

Do you prefer amended commit or should I split into more commits so that you squash it when it's done?

Reply:
Ah, java switch throws NPE when argument is null.